arraycontext/container/traversal.py

```diff
@@ -27,6 +27,8 @@
 ~~~~~~~~~~~~~~~~
 .. autofunction:: from_numpy
 .. autofunction:: to_numpy
+.. autofunction:: flatten_to_numpy
```

Comment:
> To-numpy conversion is eager by necessity (since numpy wants _data_), so doing that conversion as late and in as big a chunk as possible makes sense to me.

Good catch. I agree with this completely. So should the function just flatten all the data into a single device array then? I would really prefer that the result returned from `flatten_to_{...}` _not_ be an array container.

Reply:
Yes, flatten to a single device array makes sense to me.

Reply:
Gave this a try in ed60412.

Unfortunately, it has a big hole in it at the moment, as I'm not sure how to do the `ravel` and `reshape` in a vaguely generic manner. Suggestions?

On the bright side, it works for all the nicely C-contiguous arrays in the tests! :grin:

Reply:
Added a (hopefully) more informative `NotImplementedError` in dd75ade.
Also added some more tests in a0c9807 to check 2D arrays and `DOFArray`s with multiple groups and it seems to work.